homeassistant/components/xiaomi_miio/vacuum.py

```diff
@@ -116,110 +121,128 @@
 
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the Xiaomi vacuum cleaner robot platform."""
-    if DATA_KEY not in hass.data:
-        hass.data[DATA_KEY] = {}
+    """Import Miio configuration from YAML."""
+    _LOGGER.warning(
+        "Loading Xiaomi Miio Vacuum via platform setup is deprecated. Please remove it from your configuration."
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
 
-    host = config[CONF_HOST]
-    token = config[CONF_TOKEN]
-    name = config[CONF_NAME]
 
-    # Create handler
-    _LOGGER.info("Initializing with host %s (token %s...)", host, token[:5])
-    vacuum = Vacuum(host, token)
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Xiaomi vacuum cleaner robot from a config entry."""
+    entities = []
 
-    mirobo = MiroboVacuum(name, vacuum)
-    hass.data[DATA_KEY][host] = mirobo
+    if config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
+        if DATA_KEY not in hass.data:
+            hass.data[DATA_KEY] = {}
 
-    async_add_entities([mirobo], update_before_add=True)
+        host = config_entry.data[CONF_HOST]
+        token = config_entry.data[CONF_TOKEN]
+        name = config_entry.title
+        unique_id = config_entry.unique_id
 
-    platform = entity_platform.current_platform.get()
+        # Create handler
+        _LOGGER.info("Initializing with host %s (token %s...)", host, token[:5])
+        vacuum = Vacuum(host, token)
 
-    platform.async_register_entity_service(
-        SERVICE_START_REMOTE_CONTROL,
-        {},
-        MiroboVacuum.async_remote_control_start.__name__,
-    )
+        mirobo = MiroboVacuum(name, vacuum, config_entry, unique_id)
+        hass.data[DATA_KEY][host] = mirobo
```

Comment:
This isn't used.

Reply:
I actually did not change any of that part of the code (just indented it under the if statement).
Will have a look, maybe it is used in some service, just a moment

Reply:
I removed it, hopefully it does not break anything, I could not find anything where it was used.